### ISSUE TITLE
Certain source artist commentaries not being parsed correctly

### ISSUE THREAD

**Issue by BrokenEagle** [danbooru/danbooru#3719]
I noticed this the other day with one of my uploads. ([post #3117919](https://danbooru.donmai.us/posts/3117919))

Checking the [JSON response](https://danbooru.donmai.us/source.json?url=https%3A%2F%2Ftwitter.com%2Fzarameru_%2Fstatus%2F926475901197369349) shows the difference:

```json
  "artist_commentary": {
    "title": "",
    "description": "お題箱（https:\/\/t.co\/wbNeAVTHd9）から！サンダースのケイさん！🇺🇸 https:\/\/t.co\/h8gFN6NF77",
    "dtext_title": "",
    "dtext_description": "お題箱（"
  }
```
So the original description is still being correctly grabbed, but it isn't being parsed correctly by the DText.  I'm not sure, but maybe it's getting hung up on that Unicode parentheses? (\uFF08)

User Unbreakable also reported another misparsing from a different source. ([post #3127099](https://danbooru.donmai.us/posts/3127099))

Checking the [JSON response](https://danbooru.donmai.us/source.json?url=https%3A%2F%2Fwww.pixiv.net%2Fmember_illust.php%3Fmode%3Dmedium%26illust_id%3D68781016&ref=https%3A%2F%2Fwww.pixiv.net%2Fmember_illust.php%3Fmode%3Dmedium%26illust_id%3D68781016):

```json
  "artist_commentary": {
    "title": "しゅきかーん！いーしょーにーあーそーぼうーよー！",
    "description": "能義たか好先生(https:\/\/www.pixiv.net\/member.php?id=1067404)描き下ろしのイラストです～(*>_<*)ﾉ\r\n\r\nよろしくお願いします。\r\n\r\nhttps:\/\/nanodes.booth.pm\/items\/811631",
    "dtext_title": "しゅきかーん！いーしょーにーあーそーぼうーよー！",
    "dtext_description": "能義たか好先生(\"user\/1067404\":[https:\/\/www.pixiv.net\/member.php?id=1067404] \"»\":[\/artists?search[url_matches]=https://www.pixiv.net/member.php?id=1067404])描き下ろしのイラストです～(*>_\nよろしくお願いします。\n\nhttps:\/\/nanodes.booth.pm\/items\/811631"
  }
```

It's not easy to see the difference, but Unbreakable pointed out that it's the latter half of the emoticon ``(*>_<*)ﾉ``  that is missing.  I'm not sure what's causing this one to trip up though.  Perhaps it's the Unicode tilde that precedes it?

**Comment by Bapabooiee** [comment]
There's also a problem with fetching the commentary for [post #3230090](https://danbooru.donmai.us/posts/3230090).

It only returns

>政長さん（
>うーん、かわいい。

When it should instead return something like

>政長さん（[twitter.com/naga_masanaga](https://twitter.com/naga_masanaga)）の藍様線画を塗ってましたあ。
>うーん、かわいい。

This can be reproduced on the live site by hitting "Add commentary", manually clearing all the fields, and hitting Fetch.

**Comment by evazion** [comment]
Fixed the Twitter case.

The Pixiv case is because the `<*` in `(*>_<*)ﾉ` is being misinterpreted as the start of an html tag.

**Comment by evazion** [comment]
Another case: https://danbooru.donmai.us/posts/3239877 (NSFW).

Here the Tumblr commentary contains an `<h2>` element with a linebreak that isn't handled properly:

```
h2. [i][b]"PATREON":[https://www.patreon.com/twistedgrim]
"TWITTER":[https://twitter.com/TGnsfw][/b][/i]
```

**Comment by Bapabooiee** [comment]
[post #3248265](https://danbooru.donmai.us/posts/3248265) would be another example.

It returned

>新しいおともだち
>#けものフレンズ

When it should've instead returned something like:
>新しいおともだち
>[＃けものフレンズ](https://twitter.com/hashtag/%E3%81%91%E3%82%82%E3%81%AE%E3%83%95%E3%83%AC%E3%83%B3%E3%82%BA?src=hash)

I'm guessing it has something to do with the link in the Tweet using a full URL instead of going through the usual `t.co` domain.

**Comment by evazion** [comment]
It turns out the hashtag in that tweet contained fullwidth number sign character (＃, U+FF03) rather than a regular number sign (#, U+0023).

**Comment by evazion** [comment]
https://danbooru.donmai.us/posts/3264534. A trailing `】` character got included in the link.

**Comment by evazion** [comment]
> It seems the ideographic space is not properly detected as a space so the first link consumes the entire second link. I fixed this case by adding brackets around the URLs, but it seems incorrect.

The bug with ideographic spaces being included in links was fixed in 62160619da. c42e139e7f fixed commentaries to normalize ideographic spaces to regular spaces for most sites, and 6282dea9f fixed it specifically for Pixiv Sketch. 

> https://danbooru.donmai.us/posts/3264534. A trailing `】` character got included in the link.

Fixed in 62160619da.

> Here the Tumblr commentary contains an <h2> element with a linebreak that isn't handled properly

Fixed in c42e139e7f.
